parquet-column/src/main/java/parquet/filter2/recordlevel/IncrementallyUpdatedFilterPredicate.java

```diff
@@ -61,6 +64,24 @@
 
     private boolean result = false;
     private boolean isKnown = false;
+    private BitSet dictionaryResults;
+
+    /** Populates array of BitSet based on whether values pass filter */
+    void setDictionary(Dictionary dictionary) {
+      dictionaryResults = new BitSet(dictionary.getMaxId()+1);
+      for (int i = 0, end = dictionary.getMaxId()+1; i < end; i++) {
+        dictionaryResults.set(i, update(dictionary, i));
+      }
+    }
+
+    /** Update based on element from dictionary */
+    void updateFromDictionary(int dictionaryId) {
+      setResult(dictionaryResults.get(dictionaryId));
+    }
+
+    /** Does the element in the dictionary pass the filter */
+    abstract protected boolean update(Dictionary dictionary, int dictionaryId);
```

Comment:
I'm looking at the API: in the examples below the callee either gets the value from the dictionary or ignores the dictionary. It sounds like we could make the specialization directly on the setDictionary call.
this is similar to the dictionary support in the converter: we look at the dictionary as a whole (isNull will always return false). If the implementation needs to lookup the value in dictionary we may as well directly call the update(type) corresponding method (That's what happens in converters). It does not need to go one level below.